src/collect/perf/collect.c

```diff
@@ -66,7 +66,7 @@
 #define SYSFS_PT_TYPE   "/sys/bus/event_source/devices/intel_pt/type"
 #define MAX_PT_TYPE_STR 8
 
-#define MAX_OPEN_PERF_TRIES  20000
+#define MAX_OPEN_PERF_TRIES  50000
```

Comment:
I would suggest that adding a `pthread_yield` or `sched_yield` (or even just "sleep 100ms" or similar!) wherever this loop is will be more robust: it will make it much more likely that the other thread acquiring the lock will succeed before we try again.

Reply:
We can do, although I thought yielding was kind of frowned upon? (don't ask me why, I have a vague recollection of reading something on a mailing list)

Reply:
(btw, there's already a sleep: https://github.com/ykjit/hwtracer/blob/978aa7f88d381de1e283c5fcaae451457f7db647/src/collect/perf/collect.c#L70)

Reply:
Yielding doesn't always do what people expect (e.g. in spinlocks).

`1000 * 30` nanoseconds is 1/33333th of a second -- under such circumstances `sleep` might not even do anything! I'm fine with sleeping, but let's make it least 1/100th of a second.

Reply:
Fine by me. I don't remember where these numbers even came from TBH.

Reply:
Personally I'd use at least 1/100th of a second, perhaps as much as 1/10th. In the grand scheme of things, small delays like that don't really slow down the test suite to any noticeable degree.